ui/app/js/controllers/permission_scheduled_change_new_controller.js

```diff
@@ -0,0 +1,216 @@
+/*global sweetAlert swal */
+angular.module('app').controller('NewPermissionScheduledChangeCtrl',
+function ($scope, $modalInstance, CSRF, Permissions, scheduled_changes, sc) {
+
+  $scope.loading = true;
+  $scope.scheduled_changes = scheduled_changes;
+  $scope.scheduledpermissions = [];
+  $scope.currentItemTab = 1;
+  $scope.is_edit = true;
+  $scope.sc = angular.copy(sc);
+  $scope.permission = {
+    permission: '',
+    options: ''
+  };
+  $scope.errors = {
+    permissions: {}
+  };
+
+  $scope.sc.permissions = [];
+  Permissions.getUserPermissions(sc.username)
+  .then(function(permissions) {
+    _.forEach(permissions, function(p) {
+      if (p.options) {
+        p.options = JSON.stringify(p['options']);
+      }
+    });
+    $scope.sc.permissions = permissions;
+    // console.log('$scope.sc.permissions');
+    // console.dbg($scope.sc.permissions);
+    $scope.loading = false;
+  });
+
+
+  $scope.saving = false;
+
+
+
+
+  $scope.addScheduledPermission = function() {
+  if($scope.permission.permission) {
```

Comment:
I see that this controller is used by different modals, and that you've got addScheduledPermission for dealing with the modal that opens from the Permissions page, and addScheduledUser for dealing with the modal that opens from the Scheduled Changes page. There's a lot of duplication between them, and they both have the same job (scheduled a change for a new permission) - is there a way to combine them? If not, I think you should use a different controller for the modal that opens from the Scheduled Changes page - it's confusing to have two different handlers that both add a new scheduled change, but from different data sources.